pkgs/tools/system/zram-generator/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, ronn
+, systemd
+, kmod
+}:
+
+stdenv.mkDerivation rec {
+  pname = "zram-generator";
+  version = "1.1.2";
+
+  src = fetchFromGitHub {
+    owner = "systemd";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-n+ZOWU+sPq9DcHgzQWTxxfMmiz239qdetXypqdy33cM=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  # RFE: Include Cargo.lock in sources
+  # https://github.com/systemd/zram-generator/issues/65
+  postPatch = ''
+    cp ${./Cargo.lock} Cargo.lock
+    substituteInPlace src/generator.rs \
+      --replace 'Command::new("systemd-detect-virt")' 'Command::new("${systemd}/bin/systemd-detect-virt")' \
+      --replace 'Command::new("modprobe")' 'Command::new("${kmod}/bin/modprobe")'
+  '';
+
+  nativeBuildInputs = with rustPlatform; [
+    cargoSetupHook
+    rust.cargo
+    rust.rustc
+    pkg-config
+    ronn
+  ];
+
+  buildInputs = [
+    systemd
+  ];
+
+  makeFlags = [
+    "PREFIX=$(out)"
```

Comment:
iirc it's preferred to use `${placeholder "out"}` instead of `$(out)`

Reply:
What's the rationale?

Reply:
from what I remember, `$(out)` works because make reads this an an environment variable and `out` is set in the environment. `placeholder` is a nix language feature, so we're not relying on make behavior. or something like that.

idk if there's a consensus on this.